README.md

```diff
@@ -1,110 +1,108 @@
--->
-[![Incubating (Not yet consumable)](https://img.shields.io/badge/status-Incubating%20(Not%20yet%20consumable)-red)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
-[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-module-template?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/releases/latest)
+## IAM Service ID API key Secrets Manager module
+
+[![Graduated (Supported)](https://img.shields.io/badge/Status-Graduated%20(Supported)-brightgreen)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
+[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module/releases/latest)
 [![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit&logoColor=white)](https://github.com/pre-commit/pre-commit)
 [![Renovate enabled](https://img.shields.io/badge/renovate-enabled-brightgreen.svg)](https://renovatebot.com/)
 [![semantic-release](https://img.shields.io/badge/%20%20%F0%9F%93%A6%F0%9F%9A%80-semantic--release-e10079.svg)](https://github.com/semantic-release/semantic-release)
 
-<!-- Add a description of module(s) in this repo -->
-TODO: Replace me with description of the module(s) in this repo
+Use this module to creates a service ID API key (associated with an existing service ID) and stores it as a dynamic secret (existing or new) in an existing IBM Secrets Manager instance.
+
+Dynamic secrets, unlike (arbitrary) static secrets, create IAM service ID API key credentials that are automatically rotated by Secrets Manager engine every time the secret payload is read or accessed when the lease duration is met. The credential changes are controlled by two input variables that you specify when you create the secret: `sm_iam_secret_ttl` (sometimes referred to as _time to live_, _TTL_, or _lease duration_) and `sm_iam_secret_api_key_persistence` (sometimes referred to as _reuse IAM credentials_).
 
+In addition to rotation when the TTL or lease expires, you can create a rotation policy to trigger rotation before the leasing time expires. Triggering rotation before expiration provides transition time between the older and the new credentials. You enable this setting by setting `sm_iam_secret_auto_rotation` to `true` and configuring the related rotation variables `sm_iam_secret_auto_rotation_unit` and `sm_iam_secret_auto_rotation_interval`.
 
-<!-- Below content is automatically populated via pre-commit hook -->
 <!-- BEGIN OVERVIEW HOOK -->
 ## Overview
-* [terraform-ibm-module-template](#terraform-ibm-module-template)
+* [terraform-ibm-iam-serviceid-apikey-secrets-manager-module](#terraform-ibm-iam-serviceid-apikey-secrets-manager-module)
```

Comment:
re-run the hook as the repo has been renamed

Reply:
@ocofaigh , I have change the repo name in .github/settings.yml : `terraform-ibm-iam-serviceid-apikey-secrets-manager-module` -> `terraform-ibm-iam-serviceid-apikey-secrets-manager` , But its not populating overview hook with new name.

Reply:
@ocofaigh , here after modification receiving this error:



```
Add module repository to go.mod..........................................Failed
- hook id: go-mod-module-repository
- files were modified by this hook


current value: module github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module
new value    : module github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager

warning: If repository name has changed, then update 'remote.origin.url' locally by running 'git remote set-url origin new_repo_url' or re-clone the repo using the new repo name.
```
But the repo url is github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module